src/screens/LearningScreen/components/FlashCardContainer.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+import { View, Button, Text } from 'react-native';
+
+import { getUserData } from '../../../utils/storage';
+import {
+  getRandomWord,
+  handleCorrectOrIncorrect,
+  getNewWordLevel,
+  LIMITS,
+  LEVELS,
+} from '../../../utils/spacing';
+import FlashCard from './FlashCard';
+
+const handleRevealButtonPress = (setRevealed) => {
+  setRevealed((flipped) => !flipped);
+};
+
+const handleExpressionsButtonPress = (setExpanded) => {
+  setExpanded((isExpanded) => !isExpanded);
+};
+
+const FlashCardContainer = () => {
+  const [list, setList] = useState({});
+  const [wordInfo, setWordInfo] = useState({});
+  const [isExpanded, setExpanded] = useState(false);
+  const [isRevealed, setRevealed] = useState(false);
+  const [currentLevel, setCurrentLevel] = useState('one');
+  const [numQuizzed, setNumQuizzed] = useState(0);
+
+  const handleCorrectOrIncorrectButtonPress = (isCorrect) => {
+    const temp = handleCorrectOrIncorrect(
+      isCorrect,
+      wordInfo,
+      list,
+      currentLevel,
+    );
+
+    // EFFECTS
+    setList(temp);
+    setNumQuizzed((quizzed) => quizzed + 1);
+    setRevealed(false);
+    setExpanded(false);
+  };
+
+  useEffect(() => {
+    let mounted = true;
+
+    getUserData().then((userData) => {
+      if (mounted) {
+        setList(userData.wordList);
+      }
+    });
+
+    return () => {
+      mounted = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!list || !currentLevel || !list[currentLevel]) {
+      return;
+    }
+
+    const limit = LIMITS[LEVELS.indexOf(currentLevel)];
+
+    if (!Object.keys(list[currentLevel]).length || numQuizzed >= limit) {
```

Comment:
Conditionals like this one that have "complex" expressions can be more easily read and understood if you pull each condition out into a named variable. Up to your judgement as to when you think this applies - but in general try to consider where another developer might stumble in reading through the logic-heavy parts of your application.

```js
const currentLevelListEmpty = !Object.keys(list[currentLevel]).length
const quizzedCountPastLimit = numQuizzed >= limit

if (currentLevelListEmpty || quizzedCountPastLimit) {
  ..
}
```

Reply:
That makes sense. In this case, does it make sense to do that for something like `quizzedCountPastLimit` variable? Since I broke out `limit` into its own variable it seems like `numQuizzed >= limit` is just as clear. 

^ That's one thing that I definitely need to work on, is knowing when something is worth abstracting or if it makes enough sense on its own.